### DISCUSSION TITLE
GitHub Copilot 和 idea 的 SimilarChunksWithPaths 实现讨论

### DISCUSSION THREAD

**Discussion by c913168497** [unit-mesh/auto-dev#180] [Ideas]
有大佬能基于 SimilarChunksWithPaths 这个类的 相似度 算法提取提示词，进一步优化吗？

**Comment by phodal** [comment]
IDEA 的 SimilarChunksWithPaths 理论上是参考的是 GItHub Copilot 的。所以，两个算法基本上是一样的，优化空间应该是有的，但是我对补全的研究兴趣不大…… —— 主要还是缺少高性能的模型 API 。

但是与相似算法相比的话，AutoDev  更推荐用的是相关类来做测试生成等场景。

**Reply by phodal** [reply]
VSCode  版本的实现，应该会更好一点：https://github.com/unit-mesh/auto-dev-vscode/blob/master/src/code-search/similar/SimilarChunkTokenizer.ts

**Reply by c913168497** [reply]
你这个看起来很不错的样子，我试试 迁移到目前的插件中~ 感谢~

**Reply by c913168497** [reply]
我迁移成功了
![image](https://github.com/unit-mesh/auto-dev/assets/23418458/c53f9ff7-665e-49e8-aa6a-88330af383c8)
但是我暂时不知道它的优势在哪里，使用上来说，它处理的信息更精细，但是产生的内容丰富度有所降低

**Reply by phodal** [reply]
> 我迁移成功了
> 
> ![image](https://github.com/unit-mesh/auto-dev/assets/23418458/c53f9ff7-665e-49e8-aa6a-88330af383c8)
> 
> 但是我暂时不知道它的优势在哪里，使用上来说，它处理的信息更精细，但是产生的内容丰富度有所降低

哈哈哈，可以来 PR

**Reply by c913168497** [reply]
大佬，我使用了一下，我感觉你原来那个版本还好些，不做分词，也不做停止词，直接提交整体结果，由 大模型去做 这些事情，生成的结果反而更丰富

**Reply by phodal** [reply]
> 大佬，我使用了一下，我感觉你原来那个版本还好些，不做分词，也不做停止词，直接提交整体结果，由 大模型去做 这些事情，生成的结果反而更丰富

新版本，理论上是要做微调的

**Comment by c913168497** [comment]
OK，我看看哈，还有一个小小的建议 ![image](https://github.com/unit-mesh/auto-dev/assets/23418458/90a06ac8-8537-4ef4-b0f1-91ceeb657136)

**Reply by phodal** [reply]
可以直接来  PR  改

**Reply by c913168497** [reply]
好的
